### PR TITLE
feat: update SimpleEnvironmentSelector for OpenGraph - BED-7562

### PR DESCRIPTION
**Pull request by TheNando**
## Description

* **New Features**
  * Environment selector now renders environments dynamically with aggregated type entries, updated icons, tooltips, and enhanced filtering and search (status, risk level, platform).
  * **Note** that SimpleEnvironmentSelector will continue to exclude OpenGraph environments until filter is updated.
  * New utility helpers for environment filtering, mapping, sorting and a numeric range helper.

* **Bug Fixes**
  * Fixed delete button styling when disabled.

* **Tests**
  * Added comprehensive tests and mock environment data covering filtering, aggregation, thresholds and edge cases.

## Motivation and Context

Resolves BED-7562

## How Has This Been Tested?

Updated and additional unit tests

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have met the contributing prerequisites
  - Assigned myself to this PR
  - Added the appropriate labels
  - Associated an issue: https://github.com/SpecterOps/BloodHound/issues/672
  - Read the Contributing guide: https://github.com/SpecterOps/BloodHound/wiki/Contributing
- [x] I have ensured that related documentation is up-to-date
  - Open API docs
  - Code comments (GoDocs / JSDocs)
- [x] I have followed proper test practices
  - Added/updated tests to cover my changes
  - All new and existing tests passed

## Summary by CodeRabbit

* **New Features**
  * Dynamic platform entries and improved environment filtering/search for more accurate environment selection.

* **Tests**
  * Added extensive tests for environment filtering, aggregation checks, and selector behavior; updated existing tests and mock data.

* **Refactor**
  * Simplified environment selector internals and unified selection handlers for more consistent UX.

* **Chores**
  * Export reorganization and a small numeric utility added; minor UI styling tweaks for disabled actions.

### CODE REVIEW

File: cmd/api/src/api/v2/azure.go [hunk 1]
```diff
@@ -427,7 +427,7 @@ func (s *Resources) GetAZEntity(response http.ResponseWriter, request *http.Requ
 		api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusBadRequest, fmt.Sprintf("query parameter %s is required", objectIDQueryParameterName), request), response)
 	} else if azKind, err := azEntityParamToKind(entityType); err != nil {
 		slog.WarnContext(request.Context(), "Could not determine AZ type from entityType request var",
-			slog.String("entityType", entityType),
+			slog.String("entity_type", entityType),
```
Comment (TheNando) [inline cmd/api/src/api/v2/azure.go:430]: From `prepare-for-codereview`.

File: packages/javascript/bh-shared-ui/src/components/SimpleEnvironmentSelector/SimpleEnvironmentSelector.test.tsx [hunk 2]
```diff
@@ -45,226 +46,13 @@ beforeAll(() => server.listen());
 afterEach(() => server.resetHandlers());
 afterAll(() => server.close());
 
-const testDomains = [
```
Comment (TheNando) [inline packages/javascript/bh-shared-ui/src/components/SimpleEnvironmentSelector/SimpleEnvironmentSelector.test.tsx:48]: We had a couple of these arrays. Consolidated and moved into a mock lib. The consolidation means that many tests needed to be updated.

File: packages/javascript/bh-shared-ui/src/components/SimpleEnvironmentSelector/SimpleEnvironmentSelector.tsx [hunk 3]
```diff
@@ -69,57 +73,52 @@ const SimpleEnvironmentSelector: React.FC<{
 }> = ({ selected, align = 'start', errorMessage = '', variant, onSelect = () => {} }) => {
     const [open, setOpen] = useState<boolean>(false);
     const [searchInput, setSearchInput] = useState<string>('');
-    const { data, isLoading, isError } = useAvailableEnvironments();
+    const { data: availableEnvironments, isLoading, isError } = useAvailableEnvironments();
     const { isPrivilegeZonesPage } = usePZPathParams();
 
+    const environmentInfo = getOpenGraphEnvironmentInfoMap(availableEnvironments);
+
+    const filteredEnvironments = useMemo(
```
Comment (TheNando) [inline packages/javascript/bh-shared-ui/src/components/SimpleEnvironmentSelector/SimpleEnvironmentSelector.tsx:81]: Updated so the SimpleEnvironmentSelector sources environments from the same source as full EnvironmentSelector, which supports OpenGraph user-provided environments. There are still some missing requirements for full OG integration which have not yet been discussed, so a filter is applied such that the UX is unchanged.

File: packages/javascript/bh-shared-ui/src/components/SimpleEnvironmentSelector/SimpleEnvironmentSelector.tsx [hunk 4]
```diff
@@ -69,57 +73,52 @@ const SimpleEnvironmentSelector: React.FC<{
...
+    );
+
     const handleClose = () => setOpen(false);
 
     const handleOpenChange: (open: boolean) => void = (open) => setOpen(open);
 
     const handleChange: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement> = (e) =>
         setSearchInput(e.target.value);
 
-    const disableADPlatform = useMemo(() => {
```
Comment (TheNando) [inline packages/javascript/bh-shared-ui/src/components/SimpleEnvironmentSelector/SimpleEnvironmentSelector.tsx:82]: This was all generalized for OpenGraph support.

File: packages/javascript/bh-shared-ui/src/components/SimpleEnvironmentSelector/SimpleEnvironmentSelector.tsx [hunk 5]
```diff
@@ -69,57 +73,52 @@ const SimpleEnvironmentSelector: React.FC<{
...
 
     if (isError) return <Alert severity='error'>{errorMessage}</Alert>;
 
-    const filteredEnvironments = data?.filter(
-        (environment: Environment) =>
-            environment.name.toLowerCase().includes(searchInput.toLowerCase()) && environment.collected
-    );
-
-    const selectedEnvironmentName = selectedText(selected, data, isPrivilegeZonesPage);
+    const selectedEnvironmentName = selectedText(selected, availableEnvironments, isPrivilegeZonesPage);
```
Comment (TheNando) [inline packages/javascript/bh-shared-ui/src/components/SimpleEnvironmentSelector/SimpleEnvironmentSelector.tsx:121]: selectedText will need to be updated for full OG support.

File: packages/javascript/bh-shared-ui/src/components/SimpleEnvironmentSelector/SimpleEnvironmentSelector.tsx [hunk 6]
```diff
@@ -144,76 +143,54 @@ const SimpleEnvironmentSelector: React.FC<{
                         data-testid={'data-quality_context-selector-search'}
                     />
                 </div>
-                <ul>
+                <ul className='border-b border-neutral-light-5 pb-2 mb-2'>
```
Comment (TheNando) [inline packages/javascript/bh-shared-ui/src/components/SimpleEnvironmentSelector/SimpleEnvironmentSelector.tsx:146]: Bottom border now handled as bottom of parent rather than bottom of last child.

File: packages/javascript/bh-shared-ui/src/components/SimpleEnvironmentSelector/SimpleEnvironmentSelector.tsx [hunk 7]
```diff
@@ -144,76 +143,54 @@ const SimpleEnvironmentSelector: React.FC<{
...
+                                variant={'text'}>
+                                {environmentInfo[type]?.aggregationDisplayName}
+                                <FontAwesomeIcon icon={environmentInfo[type]?.icon} size='sm' />
+                            </Button>
+                        </li>
+                    ))}
                 </ul>
                 <ul className='max-h-80 overflow-y-auto'>
-                    {filteredEnvironments
-                        ?.sort((a: Environment, b: Environment) => {
```
Comment (TheNando) [inline packages/javascript/bh-shared-ui/src/components/SimpleEnvironmentSelector/SimpleEnvironmentSelector.tsx:181]: Sorted as part of memoization.

File: packages/javascript/bh-shared-ui/src/utils/environments.test.ts
Comment (TheNando) [inline packages/javascript/bh-shared-ui/src/utils/environments.test.ts:1]: Migrated from BHE app to shared lib.

File: packages/javascript/bh-shared-ui/src/utils/environments.ts
Comment (TheNando) [inline packages/javascript/bh-shared-ui/src/utils/environments.ts:1]: Migrated from BHE app to shared lib.

File: packages/javascript/bh-shared-ui/src/utils/environments.ts [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+
+const collectedFilterKeys = ['yes', 'no'] as const;
+const collectedKeyBoolMap = {
+    yes: true,
+    no: false,
+} as const;
+
+const riskFilterKeys = ['critical', 'high', 'moderate', 'low'] as const;
+
+export function filterAndSearchEnvironments(
```
Comment (TheNando) [inline packages/javascript/bh-shared-ui/src/utils/environments.ts:85]: I added more granularity to the filter types and did some renames to make it more concise.

File: packages/javascript/bh-shared-ui/src/utils/icons.ts [hunk 9]
```diff
@@ -234,7 +235,7 @@ export const UNKNOWN_ICON: IconInfo = {
     color: '#FFFFFF',
 };
 
-export const OPEN_GRAPH_ENVIRONMENT = 'circle-nodes';
+export const OPEN_GRAPH_ENVIRONMENT = faCircleNodes;
```
Comment (TheNando) [inline packages/javascript/bh-shared-ui/src/utils/icons.ts]: Now can be used consistency in place of other icons.

File: packages/javascript/bh-shared-ui/src/utils/number.ts [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+/** Returns true if value is greater than min and less than or equal to max */
+export function inRange(value: number, min: number, max: number) {
```
Comment (TheNando) [inline packages/javascript/bh-shared-ui/src/utils/number.ts:18]: Migrated from environments module.

File: packages/javascript/bh-shared-ui/src/views/OpenGraphManagement/ActiveExtensionsCard.test.tsx [hunk 11]
```diff
@@ -27,7 +28,7 @@ import {
     NO_SEARCH_RESULTS_MESSAGE,
 } from './ActiveExtensionsCard';
 
-const addNotificationSpy = vi.fn();
+const addNotificationSpy = vi.hoisted(() => vi.fn());
```
Comment (TheNando) [inline packages/javascript/bh-shared-ui/src/views/OpenGraphManagement/ActiveExtensionsCard.test.tsx:31]: The `hoisted` prevents race conditions for spy creation.

File: packages/javascript/bh-shared-ui/src/views/OpenGraphManagement/ActiveExtensionsCard.test.tsx [hunk 12]
```diff
@@ -278,7 +288,7 @@ describe('ActiveExtensionsCard', () => {
     });
 
     it('calls delete mutation when confirm button is clicked with correct input', async () => {
-        const deleteExtensionSpy = vi.spyOn(apiClient, 'deleteExtension').mockResolvedValue({} as any);
```
Comment (TheNando) [inline packages/javascript/bh-shared-ui/src/views/OpenGraphManagement/ActiveExtensionsCard.test.tsx:281]: More explicit type.

File: packages/javascript/bh-shared-ui/src/views/OpenGraphManagement/DeleteExtensionButton.tsx [hunk 13]
```diff
@@ -26,10 +26,10 @@ import {
 } from '@bloodhoundenterprise/doodleui';
 import { type Extension } from 'js-client-library';
 import { FC, useState } from 'react';
-import { cn, ConditionalTooltip } from '../..';
-import { AppIcon } from '../../components';
+import { AppIcon, ConditionalTooltip } from '../../components';
 import { useDeleteExtension } from '../../hooks';
 import { DEFAULT_NOTIFICATION, ERROR_NOTIFICATION, useNotifications } from '../../providers';
+import { cn } from '../../utils/theme';
```
Comment (TheNando) [inline packages/javascript/bh-shared-ui/src/views/OpenGraphManagement/DeleteExtensionButton.tsx:32]: The `../..` was causing a circular dep import warning.

File: packages/javascript/bh-shared-ui/src/views/OpenGraphManagement/DeleteExtensionButton.tsx [hunk 14]
```diff
@@ -109,7 +109,7 @@ export const DeleteExtensionButton: FC<{ extension: Extension }> = ({ extension
             <ConditionalTooltip condition={isUndeletable} tooltip='Built-in extensions cannot be deleted.'>
                 <button
                     aria-label={`Delete ${extensionName}`}
-                    className={cn('cursor-pointer', { 'opacity-50': isUndeletable })}
+                    className={cn({ 'cursor-pointer': !isUndeletable, 'opacity-50 cursor-not-allowed': isUndeletable })}
```
Comment (TheNando) [inline packages/javascript/bh-shared-ui/src/views/OpenGraphManagement/DeleteExtensionButton.tsx:112]: Change the cursor to indicate the option is disabled.

File: packages/javascript/js-client-library/src/responses.ts [hunk 15]
```diff
@@ -62,10 +62,6 @@ export const knownEnvironmentTypes = ['active-directory', 'azure'] as const;
 
 export type KnownEnvironmentType = (typeof knownEnvironmentTypes)[number];
 
-export const isKnownEnvironmentType = (type?: string): type is KnownEnvironmentType => {
```
Comment (TheNando) [inline packages/javascript/js-client-library/src/responses.ts:65]: Moved to environment module.

File: packages/javascript/bh-shared-ui/src/hooks/useAvailableEnvironments/useAvailableEnvironments.tsx [hunk 16]
```diff
@@ -42,7 +44,7 @@ export function useAvailableEnvironments<T = Environment[]>(options?: QueryOptio
 
 export const useSelectedEnvironment = (
     environmentId?: Environment['id'] | null,
-    options?: Omit<QueryOptions<Environment>, 'select'>
```
Comment (TheNando) [inline packages/javascript/bh-shared-ui/src/hooks/useAvailableEnvironments/useAvailableEnvironments.tsx:45]: I need to add this back.

File: packages/javascript/bh-shared-ui/src/utils/environments.test.ts [hunk 17]
```diff
@@ -0,0 +1,363 @@
...
+
+    describe('with no filters applied', () => {
+        it('returns all environments when no filters are active', () => {
+            const result = filterAndSearchEnvironments(testEnvironments);
+            expect(result).toEqual(testEnvironments);
+        });
+    });
+
+    describe('collected filters', () => {
+        it('filters by collected=true when "yes" filter is active', () => {
```
Comment (jvacca-specterops) [inline packages/javascript/bh-shared-ui/src/utils/environments.test.ts:50]: I'm curious as to the meaning of the yes and no filters?  I'm not familiar with this part of the codebase, but maybe we can consider a more descriptive naming convention.  Could just be my own ignorance though.  In that case please ignore.

File: packages/javascript/bh-shared-ui/src/views/OpenGraphManagement/ActiveExtensionsCard.test.tsx (hunk 11 above)
Comment (jvacca-specterops) [reply packages/javascript/bh-shared-ui/src/views/OpenGraphManagement/ActiveExtensionsCard.test.tsx:31]: Very cool.

File: packages/javascript/bh-shared-ui/src/utils/environments.test.ts (hunk 17 above)
Comment (jvacca-specterops) [reply packages/javascript/bh-shared-ui/src/utils/environments.test.ts:50]: Playing around with this locally.  Having separate checkboxes for collected: yes and collected: no feels awkward.  Would a radio or a select be a better choice here?  


You're implementation is working perfectly, however.  And the checkboxes keep it visually consistent.
